frontend/packages/noobaa-storage-plugin/src/components/data-consumption-card/data-consumption-card-utils.ts

```diff
@@ -27,183 +24,156 @@ export const numberInWords = (n: number): string => {
+          'Total Logical Used Capacity',
+        ),
+        getChartData(
+          result.logicalUsage,
+          metric,
+          humanizeBinaryBytes,
+          'Total Physical Used Capacity',
+        ),
+      ];
+      legendData = [
+        {
+          name: `Total Logical Used Capacity ${getLegendData(
+            result.totalPhysicalUsage,
+            humanizeBinaryBytes,
+          )}`,
+        },
+        {
+          name: `Total Physical Used Capacity ${getLegendData(
+            result.totalLogicalUsage,
+            humanizeBinaryBytes,
+          )}`,
+        },
+      ];
+      break;
+    case 'PROVIDERS_BY_EGRESS':
+      chartData = [getChartData(result.egress, metric, humanizeBinaryBytes)];
+      legendData = chartData[0].map((dataPoint) => ({
+        name: `${dataPoint.x} ${humanizeBinaryBytes(dataPoint.y).string}`,
+      }));
+      break;
+    default:
+      chartData = [[{ x: '', y: 0, name: '' }]];
+      legendData = [{ name: '' }];
+  }
+  return { chartData, legendData };
 };
 
-const accountsLogicalUsageChartLegendData = (data: [BarChartData[]]) => {
-  const totalLU = data[0].reduce(
-    (total: number, logicalUsage: BarChartData) => total + logicalUsage.yOriginal,
-    0,
-  );
-  const totalLUH = humanizeBinaryBytes(totalLU.toFixed(1));
-  return [{ name: `Logical Used Capacity ${totalLUH.string}` }];
+export type ChartDataPoint = {
+  x: string;
+  y: number;
+  name: string;
 };
 
-const providersEgressChartLegendData = (data: [BarChartData[]]) => {
-  const result = data[0].map((egress: BarChartData) => {
-    const totalH = humanizeBinaryBytes(egress.yOriginal.toFixed(1));
-    return { name: `${egress.x}  ${totalH.string}` };
-  });
-  return result;
-};
+type ChartData = [ChartDataPoint[]] | [ChartDataPoint[], ChartDataPoint[]];
 
-export const metricsChartDataMap = {
-  [ACCOUNTS]: {
-    [BY_IOPS]: (data) => iopsChartData(data, 'account', 'read_count', 'write_count'),
-    [BY_LOGICAL_USAGE]: (data) => accountsLogicalUsageChartData(data, 'account'),
-  },
-  [PROVIDERS]: {
-    [BY_IOPS]: (data) => iopsChartData(data, 'type', 'read_count', 'write_count'),
-    [BY_PHYSICAL_VS_LOGICAL_USAGE]: (data) => providersPhysicalVsLogicalChartData(data),
-    [BY_EGRESS]: (data) => providersEgressChartData(data),
-  },
-};
+type LegendData = { name: string }[];
 
-export const metricsChartLegendDataMap = {
-  [ACCOUNTS]: {
-    [BY_IOPS]: (data) => iopsChartLegendData(data),
-    [BY_LOGICAL_USAGE]: (data) => accountsLogicalUsageChartLegendData(data),
-  },
-  [PROVIDERS]: {
-    [BY_IOPS]: (data) => iopsChartLegendData(data),
-    [BY_PHYSICAL_VS_LOGICAL_USAGE]: (data) => providersPhysicalVsLogicalChartLegendData(data),
-    [BY_EGRESS]: (data) => providersEgressChartLegendData(data),
-  },
+type QueryObject = {
+  [key: string]: string;
 };
 
-export type BarChartData = {
-  x: string;
-  y: number;
-  name: string;
-  unit: string;
-  yOriginal: number;
+type GetChartData = (
```

Comment:
```suggestion
type GetChartDataProps = (
```

Reply:
okay, I thought capitalizing would suffice

Reply:
also its not a props type, rather a function type definition.
We can rename to `GetChartDataType` if its confusing.

Reply:
but types are capitalized and function name are camel case, which I know the pattern in general